text/0000-native-link-modifiers.md

```diff
@@ -0,0 +1,232 @@
+- Feature Name: `native_link_modifiers`
+- Start Date: 2020-06-12
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Provide an extensible mechanism for tweaking linking behavior of native libraries
+both in `#[link]` attributes (`#[link(modifiers = "+foo,-bar")]`)
+and on command line (`-l static:+foo,-bar=mylib`).
+
+# Motivation
+[motivation]: #motivation
+
+Occasionally some tweaks to linking behavior of native libraries are necessary,
+and currently there's no way to apply them.
+
+For example, some static libraries may need to be linked as a whole archive
+without throwing any object files away because some symbols in them appear to be unused
+while actually having some effect. \
+This RFC introduces modifier `whole-archive` to address this.
+
+In other cases we need to link to a library located at some specific path
+or not matching the default naming conventions. \
+This RFC introduces modifier `verbatim` to pass such libraries to the linker.
+
+This RFC also reformulates the `static-nobundle` linking kind as a modifier `bundle`
+thus providing an opportunity to change the static linking default to non-bundling
+on some future edition boundary, and hopefully unblocking its stabilization.
+
+The generic syntax provides a way to add more such modifiers in the future
+without introducing new linking kinds.
```

Comment:
IMHO this section doesn't really provide that powerful of a motivation. If you've run into needing these issues before, it seems obvious, but I've personally never had to do any special linking despite having done a lot of work in C before. I feel like examples may be helpful, even if they're contrived.

Reply:
The GitHub issues linked from this PR provide some examples, I'll try to move them into the RFC.

Reply:
Done.